src/main/java/teammates/ui/automated/PrepareEmailWorkerAction.java

```diff
@@ -0,0 +1,163 @@
+        
+        String feedbackSessionName = getRequestParamValue(ParamsNames.EMAIL_FEEDBACK);
+        
+        String courseId = getRequestParamValue(ParamsNames.EMAIL_COURSE);
+        Assumption.assertNotNull(courseId);
+        
+        log.info("Prepare email worker activated for: " + HttpRequestHelper.printRequestParameters(request));
+        
+        List<EmailWrapper> emailsToBeSent = prepareEmailsToBeSent(emailType, feedbackSessionName, courseId);
+        
+        if (emailsToBeSent == null) {
+            log.severe("Emails to be sent is null for type: " + emailTypeString);
+            setErrorResponse();
+            return;
+        }
+        
+        try {
+            new EmailSender().sendEmails(emailsToBeSent);
+            doPostProcessingForSuccesfulSend(emailType, feedbackSessionName, courseId);
+        } catch (Exception e) {
+            try {
+                doPostProcessingForUnsuccesfulSend(emailType, courseId);
+            } catch (EntityDoesNotExistException ednee) {
+                log.severe("Unexpected error: " + TeammatesException.toStringWithStackTrace(ednee));
+            }
+            log.severe("Unexpected error: " + TeammatesException.toStringWithStackTrace(e));
+            setErrorResponse();
+        }
+    }
+    
+    private void doPostProcessingForSuccesfulSend(EmailType emailType, String feedbackSessionName, String courseId)
+            throws InvalidParametersException, EntityDoesNotExistException {
+        switch (emailType) {
+        case FEEDBACK_CLOSING:
+        case FEEDBACK_CLOSED:
+        case FEEDBACK_OPENING:
+        case FEEDBACK_PUBLISHED:
+        case FEEDBACK_UNPUBLISHED:
+            FeedbackSessionAttributes feedbackSession = fsLogic.getFeedbackSession(feedbackSessionName, courseId);
+            switch (emailType) {
+            case FEEDBACK_CLOSING:
+                feedbackSession.setSentClosingEmail(true);
+                break;
+            case FEEDBACK_CLOSED:
+                feedbackSession.setSentClosedEmail(true);
+                break;
+            case FEEDBACK_OPENING:
+                feedbackSession.setSentOpenEmail(true);
+                break;
+            case FEEDBACK_PUBLISHED:
+                feedbackSession.setSentPublishedEmail(true);
+                break;
+            case FEEDBACK_UNPUBLISHED:
+                feedbackSession.setSentPublishedEmail(false);
+                break;
+            default: // impossible to reach
+                break;
+            }
+            fsLogic.updateFeedbackSession(feedbackSession);
+            break;
+            
+        case PENDING_COMMENT_CLEARED:
+            frcLogic.updateFeedbackResponseCommentsSendingState(
+                    courseId, CommentSendingState.SENDING, CommentSendingState.SENT);
+            commentsLogic.updateCommentsSendingState(courseId, CommentSendingState.SENDING, CommentSendingState.SENT);
+            break;
+            
+        default:
```

Comment:
Actually I just found a better solution: decommission this class, and:
1. For feedback session opening, closing, closed, the only client class using is the cron jobs and thus the cron jobs action class becomes something like a facade. The functions can be moved there instead. Cron jobs also have the same time limit as task queues so there is no trade-off here.
2. For feedback session (un)published and pending comment cleared there may be some performance trade-off. If this is the case, then we should have three separate action objects for those actions, much like feedback session remind emails. This means there will be no abstractions at all, but no abstraction is better than unmaintainable abstraction, right?